### PR TITLE
AG-3776 - added onDragCancel

### PR DESCRIPTION
**Pull request by gportela85**
https://ag-grid.atlassian.net/browse/AG-3776
https://ag-grid.atlassian.net/browse/AG-12446

### CODE REVIEW

File: community-modules/core/src/dragAndDrop/dragService.ts [hunk 1]
```diff
@@ -177,12 +178,18 @@ export class DragService extends BeanStub implements NamedBean {
         const mouseMoveEvent = (event: MouseEvent) => this.onMouseMove(event, params.eElement);
         const mouseUpEvent = (event: MouseEvent) => this.onMouseUp(event, params.eElement);
         const contextEvent = (event: MouseEvent) => event.preventDefault();
+        const keydownEvent = (event: KeyboardEvent) => {
+            if (event.key === KeyCode.ESCAPE) {
+                this.cancelDrag(params.eElement);
+            }
```
Comment (peterjrreynolds) [inline community-modules/core/src/dragAndDrop/dragService.ts:184]: Do we need to stop the event propagating here?

File: community-modules/core/src/events.ts [hunk 2]
```diff
@@ -433,10 +435,14 @@ export interface DragStartedEvent<TData = any, TContext = any> extends AgDragEve
     type: 'dragStarted';
 }
 
-export interface DragStoppedEvent<TData = any, TContext = any> extends AgDragEvent<`dragStopped`, TData, TContext> {
+export interface DragStoppedEvent<TData = any, TContext = any> extends AgDragEvent<'dragStopped', TData, TContext> {
     type: 'dragStopped';
 }
 
+export interface DragCancelledEvent<TData = any, TContext = any> extends AgDragEvent<'dragCancelled', TData, TContext> {
+    type: 'dragCancelled';
```
Comment (peterjrreynolds) [inline community-modules/core/src/events.ts:443]: Don't need the `type` here - it's passed through the generic param.

In fact, don't need the type for the above two events either. Looks like these got left in by accident when the generic param was added.

File: community-modules/core/src/gridBodyComp/rowDragFeature.ts [hunk 3]
```diff
@@ -368,6 +383,12 @@ export class RowDragFeature extends BeanStub implements DropTarget {
                       events.onDragStop!(this.draggingToRowDragEvent('rowDragEnd', e as any));
                   }
                 : onDragStop,
+            onDragCancel: events.onDragCancel
```
Comment (peterjrreynolds) [inline community-modules/core/src/gridBodyComp/rowDragFeature.ts:386]: Could de-duplicate the code for all these events

File: community-modules/core/src/events.ts (hunk 2 above)
Comment (gportela85) [reply community-modules/core/src/events.ts:443]: fixed

File: community-modules/core/src/dragAndDrop/dragService.ts (hunk 1 above)
Comment (gportela85) [reply community-modules/core/src/dragAndDrop/dragService.ts:184]: I don't think we should stop the propagation here, we can perform whatever actions we need and still allow the customer to add their own listeners.

**Pull request merged**